src/modules/private.ts

```diff
@@ -475,12 +485,37 @@ export default class Private {
     * @signature starkware specific signature for fast-withdrawal
     * }
     */
-  createFastWithdrawal(
+  async createFastWithdrawal(
     params: PartialBy<ApiFastWithdrawal, 'clientId' | 'signature'>,
+    positionId: string,
   ): Promise<{ withdrawal: TransferResponseObject }> {
     const clientId = params.clientId || Math.random().toString().slice(2).replace(/^0+/, '');
     // TODO meet starkware specification
-    const signature = params.signature || Math.random().toString().slice(2).replace(/^0+/, '');
+
+    let signature: string | undefined = params.signature;
+    if (!signature) {
+      if (!this.starkKeyPair) {
+        throw new Error('Fast withdrawal is not signed and client was not initialized with starkPrivateKey');
+      }
+      const fact = this.starkLib.factRegistry.getTransferErc20Fact({
+        recipient: params.toAddress,
+        tokenAddress: this.starkLib.collateralToken.getAddress(),
+        tokenDecimals: this._collateralTokenDecimals,
+        humanAmount: params.creditAmount,
+        salt: nonceFromClientId(clientId),
+      });
+      const conditionalTransfer = new SignableConditionalTransfer({
+        senderPositionId: params.lpPositionId,
+        receiverPositionId: positionId,
+        receiverPublicKey: this.starkKeyPair.publicKey,
```

Comment:
This may be backwards, the conditional transfer should be received by the LP.

Compare with librarian: https://github.com/dydxprotocol/librarian/blob/master/src/controllers/api/v3/fast-withdrawals-controller.ts#L185-L193

Reply:
updated